src/lib/schedule.c

```diff
@@ -25,459 +25,95 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  *
- * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ * Author: Bartosz Kokoszko <bartoszx.kokoszko@linux.intel.com>
  */
 
-#include <stdint.h>
-#include <stddef.h>
-#include <errno.h>
-#include <sof/sof.h>
-#include <sof/lock.h>
-#include <sof/list.h>
-#include <sof/stream.h>
-#include <sof/alloc.h>
-#include <sof/debug.h>
-#include <sof/clk.h>
+/* Generic scheduler */
 #include <sof/schedule.h>
-#include <sof/work.h>
-#include <platform/timer.h>
-#include <platform/clk.h>
-#include <sof/audio/component.h>
-#include <sof/drivers/timer.h>
-#include <sof/task.h>
+#include <sof/edf_schedule.h>
+#include <sof/ll_schedule.h>
 
-struct schedule_data {
-	spinlock_t lock;
-	struct list_item list;	/* list of tasks in priority queue */
-	uint32_t clock;
-	struct work work;
+struct scheduler_ops *schedulers[SOF_TASKS_COUNT] = {
+	&schedule_edf_ops,              /* SOF_EDF_TASK */
+	&schedule_ll_ops		/* SOF_LOW_LATENCY_TASK */
 };
 
-#define SLOT_ALIGN_TRIES	10
-
-/*
- * Simple rescheduler to calculate tasks new start time and deadline if
- * prevoius deadline was missed. Tries to align at first with current task
- * timing, but will just add onto current if too far behind current.
- * XRUNs will be propagated upto the host if we have to reschedule.
- */
-static inline void edf_reschedule(struct task *task, uint64_t current)
+void schedule_task_init(struct task *task, uint32_t type, uint32_t priority,
```

Comment:
What about failure ? We probably want to return a error here.

Reply:
Changed to ```int schedule_task_int()```